ssz-rs/src/merkleization/mod.rs

```diff
@@ -129,10 +132,17 @@ fn merkleize_chunks_with_virtual_padding(
                 Ordering::Less => {
                     let focus =
                         &mut layer[parent_index * BYTES_PER_CHUNK..(i + 2) * BYTES_PER_CHUNK];
+                    // checked subtraction is unnecessary:
+                    // focus.len() = (i + 2 - parent_index) * BYTES_PER_CHUNK
+                    // and
+                    // i >= parent_index
+                    // so focus.len() >= 2 * BYTES_PER_CHUNK; qed
                     let children_index = focus.len() - 2 * BYTES_PER_CHUNK;
                     let (parent, children) = focus.split_at_mut(children_index);
                     let (left, right) = children.split_at_mut(BYTES_PER_CHUNK);
                     if parent.is_empty() {
+                        // TODO: Why not i == 0? Is there another case where the left node is used
+                        // to store the parent?
```

Comment:
I'd have to dig back into this code to answer this question; let's make an issue

Reply:
#63

Reply:
I recall now that I *think* this is here because it just makes the math simpler if we ignore the node at `i==0`

I will re-verify but basically we are just saying "its ok to waste these 32 bytes" bc the code is much easier to implement

I don't know if a more sophisticated implementation will ever be needed but you can go beyond the relatively simple Merkle code here so I didn't feel like it was that important to put that much effort into this

a benchmark and a use case is all it would take to justify improving this however :)

Reply:
if this is in fact the case ill add a clarifying comment somewhere about the data structure; if the issue is just an offset by 1 then I should also be able to refactor very easily and then skip the "wasted" bytes

Reply:
scratch all of this... 

ill move to the issue